crates/astria-conductor/tests/blackbox/soft_only.rs

```diff
@@ -333,3 +360,97 @@ async fn requests_from_later_genesis_height() {
          within 1000ms",
     );
 }
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 1)]
+async fn exits_on_sequencer_chain_id_mismatch() {
+    use astria_grpc_mock::{
+        matcher,
+        response as GrpcResponse,
+        Mock as GrpcMock,
+    };
+
+    // We have to create our own test conductor and perform mounts manually because `TestConductor`
+    // implements the `Drop` trait, which disallows us from taking ownership of its tasks and
+    // awaiting their completion.
+
+    let mock_grpc = MockGrpc::spawn().await;
+    let mock_http = wiremock::MockServer::start().await;
+
+    let config = Config {
+        celestia_node_http_url: mock_http.uri(),
+        execution_rpc_url: format!("http://{}", mock_grpc.local_addr),
+        sequencer_cometbft_url: mock_http.uri(),
+        sequencer_grpc_url: format!("http://{}", mock_grpc.local_addr),
+        execution_commit_level: CommitLevel::SoftOnly,
+        ..make_config()
+    };
+
+    let (metrics, _) = metrics::ConfigBuilder::new()
+        .set_global_recorder(false)
+        .build(&())
+        .unwrap();
+    let metrics = Box::leak(Box::new(metrics));
+
+    let conductor = {
+        let conductor = Conductor::new(config, metrics).unwrap();
+        conductor.spawn()
+    };
+
+    GrpcMock::for_rpc_given(
+        "get_genesis_info",
+        matcher::message_type::<GetGenesisInfoRequest>(),
+    )
+    .respond_with(GrpcResponse::constant_response(
+        genesis_info!(sequencer_genesis_block_height: 1,
+            celestia_block_variance: 10,),
+    ))
+    .expect(0..)
+    .mount(&mock_grpc.mock_server)
+    .await;
+
+    GrpcMock::for_rpc_given(
+        "get_commitment_state",
+        matcher::message_type::<GetCommitmentStateRequest>(),
+    )
+    .respond_with(GrpcResponse::constant_response(commitment_state!(firm: (
+            number: 1,
+            hash: [1; 64],
+            parent: [0; 64],
+        ),
+        soft: (
+            number: 1,
+            hash: [1; 64],
+            parent: [0; 64],
+        ),
+        base_celestia_height: 1,)))
+    .expect(0..)
+    .mount(&mock_grpc.mock_server)
+    .await;
+
+    mount_genesis(&mock_http, "bad_chain_id").await;
+
+    let res = conductor.await;
+    match res {
+        Ok(Ok(())) => panic!("conductor should have exited with an error, no error received"),
+        Ok(Err(e)) => {
+            let mut source = e.source();
+            while source.is_some() {
+                let err = source.unwrap();
+                if let Some(SequencerChainIdError::MismatchedSequencerChainId {
+                    expected,
+                    actual,
+                }) = err.downcast_ref::<SequencerChainIdError>()
```

Comment:
Don't test implementation details in blackbox tests.

It's fine to assert that it exits after receiving a bad chain ID.